src/daft-core/src/series/ops/date.rs

```diff
@@ -37,12 +38,80 @@ impl Series {
 
     pub fn dt_hour(&self) -> DaftResult<Self> {
         match self.data_type() {
-            DataType::Timestamp(..) => {
+            DataType::Timestamp(tu, _) => {
+                let tu = match tu {
+                    TimeUnit::Nanoseconds => TimeUnit::Nanoseconds,
+                    _ => TimeUnit::Microseconds,
```

Comment:
What is the reason for changing the time unit?

Reply:
Only microseconds and nanoseconds time units are supported for the Time dtype currently.

Reply:
https://github.com/Eventual-Inc/Daft/blob/416009138359a9d410db90819ea708b630aa7a39/src/daft-core/src/array/ops/date.rs#L186-L195

@kevinzwang  the timeunit here is used for casting i.e. changing the underlying data
also passing along the timeunit to time as it is would lead to unwanted errors

https://github.com/Eventual-Inc/Daft/blob/416009138359a9d410db90819ea708b630aa7a39/src/daft-core/src/array/ops/date.rs#L180

Reply:
Oh nevermind this was a misunderstanding on my part of what the TimeUnit input to `TimestampArray::time` was used for.